configure.ac

```diff
@@ -2333,16 +2333,24 @@ struct sockaddr_storage {
 #error dont know what to use for SIZE_T_FMT
 #endif
 
-#if (SIZEOF_OFF_T == SIZEOF_LONG)
-#define OFF_T_FMT "%ld"
-#define strtoofft(nptr, endptr, base) strtol(nptr, endptr, base)
-#elif (SIZEOF_OFF_T == SIZEOF_LONG_LONG_INT)
+#if (SIZEOF_OFF_T == SIZEOF_LONG_LONG_INT)
 #define OFF_T_FMT "%lld"
 #define strtoofft(nptr, endptr, base) strtoll(nptr, endptr, base)
+#elif (SIZEOF_OFF_T == SIZEOF_LONG)
+#define OFF_T_FMT "%ld"
+#define strtoofft(nptr, endptr, base) strtol(nptr, endptr, base)
 #else
 #error dont know what to use for OFF_T_FMT
 #endif
 
+#if (SIZEOF_TIME_T == SIZEOF_LONG_LONG_INT)
+#define TIME_T_FMT "%lld"
+#elif (SIZEOF_TIME_T == SIZEOF_LONG)
+#define TIME_T_FMT "%ld"
+#else
+#error dont know what to use for TIME_T_FMT
+#endif
+
```

Comment:
Annoyingly, changing the order here breaks the linux/gcc/x86_64 build!

I think because on this platform, SIZEOF_OFF_T == SIZEOF_LONG == SIZEOF_LONG_LONG_INT, so whichever of those conditions we put first is the one that matches, but it _really_ wants `%ld`, and complains if it gets `%lld`, even though they're the same size...

Reply:
(Same applies for TIME_T_FMT too: it really wants %ld, but all three types are the same size, so whichever is checked first wins.)

Reply:
In this case we need to properly test the format specifiers in `configure.am`. I guess what you were trying with the redefined-type test is the way to go.